### ISSUE TITLE
Setting name index column of dataframes when converting into CAS Tabes does not appear when saving as sashdat file

### ISSUE THREAD

**Issue by vinayash** [sassoftware/python-swat#46]
I think I have identified a limitation/defect when saving a CAS table as a sashdat file after being converted from a dataframe.

By default, python creates an index column (without a name) that provides a unique ID number to each observation.
For a dataset I was working with, that had no unique ID values, I first tried to convert this dataset to a dataframe using pandas, and then I tried to assign a name to the index of the dataframe.

For example, I tried assigning the index to a name 'ID', by submitting the following code:

df.set_index('ID', inplace=True).

This does create 'ID' successfully. I can even view the column interactively in the python IDE (I'm using Jupyter) when I print the dataframe.

I can also view the 'ID' column when I save to a csv file using:

import csv 
df.to_csv('filename')

Ex output: 

    ID, C1, C2
    0,   X,  Y
    1,   A,  B

The problem occurs when I save as a sashdat file. 
When I try to save the dataframe converted CAS Table with the 'ID' column to a sashdat file,
the resulting Table does not contain the newly named 'ID' column. Rather the renamed index called 'ID' appears without a name. 

Ex. output:

       C1  C2
    0  X   Y
    1  A   B

I can confirm this by loading the sashdat file and reading the table. 

Here is my methodology:
I convert the dataframe to a dictionary first using to_dict(), and then from a dictionary to a CAS Table using CasTable.from_dict().

So in summary, renaming the index of the dataframe does not appear after saving as sashdat file and loading back as a CAS Table.

Is this something that is expected, or a defect? 
Is it best practice not to rename the index of a dataframe, or does this have scope to be fixed?

**Comment by kesmit13** [comment]
Just out of curiosity, what does the dictionary look like after you do to_dict()?  Does it contain a key 'ID'?

**Comment by kesmit13** [comment]
I think the problem stems from the cycle of to_dict() and from_dict().  If you look at the following code, you'll see that when doing to_dict(), the ID column doesn't as a labeled item anymore.  So when you try to do a from_dict() on it, you just get an unlabeled ID which isn't transferred to the CAS table.  You don't really need to convert to a dict though, you should be able to just upload the dataframe using the conn.upload_frame(...) method.  I believe the index column will be uploaded correctly using that.

```
In [10]: df.to_dict()
Out[10]: {'C1': {0: 'X', 1: 'A'}, 'C2': {0: 'Y', 1: 'B'}}

In [12]: pd.DataFrame.from_dict(df.to_dict())
Out[12]:
  C1 C2
0  X  Y
1  A  B
```

**Comment by vinayash** [comment]
Kevin,

Thank you for your reply and apologies for the late response. I was out this weekend with no connection online.

I will try the upload.frame() function some time soon, and examine the behavior.

Thanks again for the help.

**Comment by kesmit13** [comment]
I'm quite sure this is an issue with the usage of `to_dict` / `from_dict`, so I'm going to close it.
